components/core/src/useManagedState.ts

```diff
@@ -1,8 +1,12 @@
-import { ChangeEvent, useRef, useState } from 'react';
+import { useRef, useState } from 'react';
 
 import { devWarning } from './devWarning';
 
-export function useManagedState(controlledValue, defaultValue, onChange) {
+export function useManagedState<Tvalue>(
+	controlledValue: Tvalue,
+	defaultValue: Tvalue,
+	onChange: Function
```

Comment:
minor: I think onChange is expecting to be something like:
`onChange: (value: any, event: Event) => any;`